src/gd_heif.c

```diff
@@ -321,13 +321,17 @@ static int _gdImageHeifCtx(gdImagePtr im, gdIOCtx *outfile, int quality, gdHeifC
 		return GD_FALSE;
 	}
 
+#if LIBHEIF_HAVE_VERSION(1, 9, 0)
```

Comment:
if the api version is available at runtime (and it seems so with your test changes), then lets do that here too.

```c
if (LIBHEIF_HAVE_VERSION(1, 9, 0) || heif_get_version_number() >= LIBHEIF_MAKE_VERSION(1, 9, 0)) {
```

Reply:
Done.